pkg/scheduler/framework/plugins/interpodaffinity/filtering.go

```diff
@@ -128,18 +127,25 @@ func (m topologyToMatchedTermCount) updateWithAffinityTerms(
 func (m topologyToMatchedTermCount) updateWithAntiAffinityTerms(terms []framework.AffinityTerm, pod *v1.Pod, nsLabels labels.Set, node *v1.Node, value int64) {
 	// Check anti-affinity terms.
 	for _, t := range terms {
+		topologyValue, ok := node.Labels[t.TopologyKey]
+		if !ok {
+			continue
+		}
 		if t.Matches(pod, nsLabels) {
-			m.update(node, t.TopologyKey, value)
+			m.update(t.TopologyKey, topologyValue, value)
 		}
 	}
 }
 
 // returns true IFF the given pod matches all the given terms.
-func podMatchesAllAffinityTerms(terms []framework.AffinityTerm, pod *v1.Pod) bool {
+func podMatchesAllAffinityTerms(node *v1.Node, terms []framework.AffinityTerm, pod *v1.Pod) bool {
 	if len(terms) == 0 {
 		return false
 	}
 	for _, t := range terms {
+		if _, ok := node.Labels[t.TopologyKey]; !ok {
+			continue
+		}
```

Comment:
Why do we need to have this change?
We've confirmed that node has all TopologyKey label in L349 - L359.

Reply:
`podMatchesAllAffinityTerms` will be called not only in `satisfyPodAffinity`, but also in `updateWithAffinityTerms` (the latter does not ensure that the label must exist)